src/network/networkd-json.c

```diff
@@ -1044,6 +1045,31 @@ static int dhcp_server_append_json(Link *link, JsonVariant **v) {
         return json_append_one(v, "DHCPServer", w);
 }
 
+static int dhcp6_client_vendor_options_append_json(Link *link, JsonVariant **v) {
+        _cleanup_(json_variant_unrefp) JsonVariant *array = NULL;
+        int r;
+
+        assert(link);
+        assert(v);
+
+        if (!link->dhcp6_lease)
+                return 0;
+
+        LIST_FOREACH(options, option, link->dhcp6_lease->vendor_options) {
+                char key[STRLEN("VENDOR_SPECIFIC_SUB_OPTION_") + DECIMAL_STR_MAX(uint16_t)];
+
+                xsprintf(key, "VENDOR_SPECIFIC_SUB_OPTION_%" PRIu16, option->option_code);
```

Comment:
We usually name keys like the following: `VendorSpecificSubOption`
Maybe, in short, `SubOption` is enough.

Reply:
Can we have VendorSubOption_XXX instead to be more precise ?

Reply:
Sounds good.

Reply:
@yuwata : As per the RFC client can receive more than one OPTION_17 in the packet and this is differentiated by the enterprise id which is unique.

Something like below:

Option 17:
   EnterpriseNumber: 314
      SubOption: 245
      SubOption: 246
      SubOption: 247
Option 17
   EnterpriseNumber: 317
      SubOption: 245
      SubOption: 246
      SubOption: 247
  
So can we have option name encoded with enterprise number to differentiate between multiple OPTION_17 like
VendorSubOption_<ENTERPRISE_ID>_<SUBOPTION_ID> (VendorSubOption_313_245)

Reply:
Ah, indeed. Then, please do not encode the enterprise ID in the option name.
How about the following?
```
"VendorSpecificInformation" : [
  {
    "EnterpriseID" : 314,
    "SubOptions" : [
      {
        "SubOptionCode" : 245,
        "SubOptionData" : "hex data"
      },
      {
        "SubOptionCode" : 246,
        "SubOptionData" : "hex data"
      },
      {
        "SubOptionCode" : 247,
        "SubOptionData" : "hex data"
      },
    ]
  },
  {
    "EnterpriseID" : 315,
    "SubOptions" : [
      {
        "SubOptionCode" : 245,
        "SubOptionData" : "hex data"
      },
      {
        "SubOptionCode" : 246,
        "SubOptionData" : "hex data"
      },
      {
        "SubOptionCode" : 247,
        "SubOptionData" : "hex data"
      },
    ]
  }
]
```

Reply:
Can it be like this ?

`"VendorSpecificInformation" : [
  {
    "EnterpriseID" : 314,
    "SubOptionCode" : 245,
    "SubOptionData" : "hex data"
   },
 ]`